### PR TITLE
Feature/us656630 bundle dependencies

### PR DESCRIPTION
**Pull request by uppoju**
## Description
Enhanced bundle dependencies tag to read bundle metadata
Supported id and guid annotations underr @reusable and @bundle
## Links
Link Type   | Link
------      | ------
Rally Issue | [us656630 ]
Func Spec   | [Func Spec]

## Checklist
- [x] Pull Request Created
- [ ] Code Review Completed
- [ ] Veracode scan results addressed
- [ ] SonarQube scan results addressed
- [ ] TPSR has been submitted (if applicable) 
- [ ] Unit tests have been added
- [ ] Integration/Functional test cases have been written and automated
- [ ] Func Spec has been updated as needed
- [ ] Rally Issue(s) are in an engineering completed state

### CODE REVIEW

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/EncassEntityBuilder.java [hunk 1]
```diff
@@ -82,21 +83,59 @@ private boolean isAnnotatedEntity(Encass encass, AnnotatedEntity annotatedEntity
...
+            if (bundleMetadataSet != null && !bundleMetadataSet.isEmpty()) {
+                for (BundleDefinedEntities bundleMetadata : bundleMetadataSet) {
+                    Collection<BundleDefinedEntities.DefaultMetadata> metadataCollection = bundleMetadata.getDefinedEntities();
+                    if (metadataCollection != null) {
+                        Optional<BundleDefinedEntities.DefaultMetadata> optionalMetadata = metadataCollection.stream().
+                                filter(metadata -> metadata.getName().equals(PathUtils.extractName(policyWithPath))
+                                        && metadata.getType().equals(EntityTypes.POLICY_TYPE)).findFirst();
+                        if (optionalMetadata.isPresent()) {
+                            if (policyId == null) {
+                                policyId = optionalMetadata.get().getId();
```
Comment (RaghuReddy) [inline config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/EncassEntityBuilder.java:100]: is this logic is not applicable to set GUID?

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/PolicyEntityBuilder.java [hunk 2]
```diff
@@ -260,16 +273,48 @@ private Encass getEncass(Bundle bundle, String name) {
...
         }
         return referenceEncass.get();
     }
 
+    private Encass getEncassFromMetadata(final Metadata metadata) {
+        Encass encass = new Encass();
+        encass.setId(metadata.getId());
+        encass.setGuid(metadata.getGuid());
+        Set<Annotation> annotations = new HashSet<>();
+        annotations.add(AnnotableEntity.REUSABLE_ANNOTATION);
```
Comment (RaghuReddy) [inline config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/PolicyEntityBuilder.java:313]: are we adding the @reusable flag to policy if it is found in another bundle file?

File: config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/EncassEntityBuilder.java (hunk 1 above)
Comment (RaghuReddy) [reply config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/EncassEntityBuilder.java:100]: correct me , this is to find the dependent policy from other modules?

Comment (uppoju) [reply config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/EncassEntityBuilder.java:100]: yes

**Pull request merged**